### PR TITLE
Throw ApiException when not enough params for function call given

### LINKED ISSUE #198: ArrayIndexOutOfBoundsException while executing a function with not enough input parameters
```console
$ ./bin/uc function call --full_name unity.default.sum --input_params 1,2
java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at io.unitycatalog.cli.utils.PythonInvoker.invokePython(PythonInvoker.java:40)
	at io.unitycatalog.cli.FunctionCli.executeFunction(FunctionCli.java:107)
	at io.unitycatalog.cli.FunctionCli.handle(FunctionCli.java:49)
	at io.unitycatalog.cli.UnityCatalogCli.main(UnityCatalogCli.java:85)
Error occurred while executing the command. Failed to execute function unity.default.sumError invoking Python script: Index 2 out of bounds for length 2
```

This exception is thrown as the function requires 3 input parameters.

```console
$ ./bin/uc function get --full_name unity.default.sum --output jsonPretty | jq '.input_params.parameters[].name'
"x"
"y"
"z"
```

### PR DESCRIPTION
**Pull request by jaceklaskowski**
Throw an `ApiException` when there is not enough required parameters given to execute (_call_) a function. Include the names of the missing params.

```
$ ./bin/uc function call --full_name unity.default.sum --input_params 1,2
io.unitycatalog.client.ApiException: Not enough parameters provided: 2, expected: 3, missing: [z]
	at io.unitycatalog.cli.utils.PythonInvoker.invokePython(PythonInvoker.java:39)
	at io.unitycatalog.cli.FunctionCli.executeFunction(FunctionCli.java:109)
	at io.unitycatalog.cli.FunctionCli.handle(FunctionCli.java:51)
	at io.unitycatalog.cli.UnityCatalogCli.main(UnityCatalogCli.java:85)
Error occurred while executing the command. Failed to execute function unity.default.sumError invoking Python script: Not enough parameters provided: 2, expected: 3, missing: [z]
```

Closes #198

Signed-off-by: Jacek Laskowski <jacek@japila.pl>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-etc/db/h2db.mv.db
```
Comment (vikrantpuppala) [inline .gitignore:1]: This file was added here as this was meant to be a starter database with prepopulated data that we would not want to edit. There are efforts to move this to a script, could we keep this until then please? #152 for reference.

File: examples/cli/src/main/java/io/unitycatalog/cli/utils/PythonInvoker.java [hunk 2]
```diff
@@ -27,17 +27,21 @@ public static String invokePython(FunctionInfo function, String scriptPath, Stri
...
 
             // Retrieve and add parameters as arguments
-            List<FunctionParameterInfo> parameters = function.getInputParams()
+            List<FunctionParameterInfo> parameters = function
+                    .getInputParams()
                     .getParameters();
             if (parameters == null || parameters.isEmpty()) {
                 throw new ApiException("Function parameters not found.");
             }
+            if (args.length < parameters.size()) {
```
Comment (vikrantpuppala) [inline examples/cli/src/main/java/io/unitycatalog/cli/utils/PythonInvoker.java:36]: Should we add a more descriptive message like which parameters were not provided?

File: examples/cli/src/main/java/io/unitycatalog/cli/utils/PythonInvoker.java [hunk 3]
```diff
@@ -27,17 +27,21 @@ public static String invokePython(FunctionInfo function, String scriptPath, Stri
...
+                throw new ApiException("Not enough parameters provided: " + args.length + ", expected: " + parameters.size());
+            }
             List<String> paramNames = new ArrayList<>();
             List<Object> argValues = new ArrayList<>();
-            int index = 0;
             for (FunctionParameterInfo param : parameters) {
+                System.out.println(">>> param=" + param);
```
Comment (vikrantpuppala) [inline examples/cli/src/main/java/io/unitycatalog/cli/utils/PythonInvoker.java:42]: Do we need this?

**Review by vikrantpuppala — COMMENTED** [review]
minor comments

**Comment by vikrantpuppala** [comment]
Could you also link an issue to this please?

**Comment by jaceklaskowski** [comment]
Hey @vikrantpuppala, can you review again, please? 🙏

**Review by vikrantpuppala — APPROVED** [review]
lgtm, @haogang to confirm and merge

**Review by haogang — APPROVED** [review]
lgtm. can you please rebase so that we can merge

**Pull request merged**